### PR TITLE
OCPBUGS-13355: Use RequestToken functions from library-go

### PR DESCRIPTION
**Pull request by stlaz**
This PR shows what needs to be done in order to make use of the shared code from https://github.com/openshift/library-go/pull/1267 for requesting tokens.

/assign @soltysh 
cc @s-urbaniak

### CODE REVIEW

**Comment by stlaz** [comment]
/reopen
still valid, I need to refresh on what I did back then and finish this PR

**Comment by stlaz** [comment]
The above PR brings the console URL back to the basic auth challenger in lib-go

**Comment by ardaguclu** [comment]
I let my 1.20 bump PR https://github.com/openshift/oc/pull/1406 is merged, but in any case we need this PR to unify request token in one package(library-go)

**Comment by stlaz** [comment]
/retest-required
payload building issues

**Comment by stlaz** [comment]
/retest-required
unrelated

File: pkg/helpers/authchallengers/negotiator_sspi.go [hunk 1]
```diff
@@ -246,19 +246,15 @@ func (s *sspiNegotiator) getPassword(domain, username string) (string, error) {
 	password := s.password
 
 	if missingPassword := len(password) == 0; missingPassword {
+		if len(s.webConsoleURL) > 0 {
```
Comment (ardaguclu) [inline pkg/helpers/authchallengers/negotiator_sspi.go:249]: nit: we are changing the logging order from;
```go
fmt.Fprintf(s.writer, "Authentication required for %s (%s)\n", s.host, domain)
fmt.Fprintf(s.writer, "Console URL: %s/console\n", s.host)
```
to
```go
fmt.Fprintf(s.writer, "Console URL: %s/console\n", s.host)
fmt.Fprintf(s.writer, "Authentication required for %s (%s)\n", s.host, domain)
```

File: pkg/helpers/tokencmd/basicauth_test.go [hunk 2]
```diff
@@ -1,371 +0,0 @@
...
-					ExpectedHandled:   true,
-					ExpectedErr:       nil,
-					ExpectedPrompt: `Authentication required for https://myhost (myrealm)
-Username: myuser
-Password: `,
-				},
-			},
-		},
-
-		"interactive challenge with default user with console url": {
```
Comment (ardaguclu) [inline pkg/helpers/tokencmd/basicauth_test.go:116]: nit: only this test does not exist in the library-go basicauth unit tests

File: pkg/helpers/tokencmd/request_token_test.go [hunk 3]
```diff
@@ -1,745 +0,0 @@
...
-
-	"github.com/RangelReale/osincli"
-
-	"k8s.io/apimachinery/pkg/util/diff"
-	restclient "k8s.io/client-go/rest"
-
-	"github.com/openshift/library-go/pkg/oauth/oauthdiscovery"
-)
-
-type unloadableNegotiator struct {
```
Comment (ardaguclu) [inline pkg/helpers/tokencmd/request_token_test.go:22]: These negotiator tests seem useful and maybe we should keep them under authchallengers in negotiate_tests.go. WDYT?

**Review by ardaguclu — COMMENTED** [review]
Code generally looks good to me and left a few comments. But I'm not sure we have test coverage for the changes in this PR. Would be better to do pre-merge test?. @zhouying7780 would you mind have a look at this one when you have cycles?

File: pkg/helpers/authchallengers/negotiator_sspi.go (hunk 1 above)
Comment (stlaz) [reply pkg/helpers/authchallengers/negotiator_sspi.go:249]: yes, I believe that this is the proper order. I'd rather not have text in between "Authentication required" and the password prompt.

File: pkg/helpers/tokencmd/request_token_test.go (hunk 3 above)
Comment (stlaz) [reply pkg/helpers/tokencmd/request_token_test.go:22]: good catch, I'll see where would be a good place for them

Comment (stlaz) [reply pkg/helpers/tokencmd/request_token_test.go:22]: the tests are now back :slightly_smiling_face:

**Comment by zhouying7780** [comment]
Test the latest code oc login with ldap user works well . @stlaz

**Pull request merged**